### ISSUE TITLE
coolprop.org doesn't support HTTPS

### ISSUE THREAD

**Issue by softwareengineerprogrammer** [CoolProp/CoolProp#2350]
Most browsers (rightfully) warn when sites don't support HTTPS and make users click through a warning dialog to confirm visiting an HTTP site

**Comment by ibell** [comment]
Yup - agreed. But getting SSL certs and so on is not free and requires expertise to implement. Want to volunteer?

**Comment by ibell** [comment]
@jowr do you have some input? Maybe you can coordinate offline?

**Comment by jowr** [comment]
I have no way of handling this. I do not own the domain and that is where the settings have to be applied. Was it you who owns the domain @Rod-Persky?

**Comment by Rod-Persky** [comment]
Hi All,

letsencrypt would work, the only requirement is to have the server setup to respond and load the certificate.

The config for DNS points to 216.105.38.10

Happy to help, but this will be something that will need to be done on the server side.

![image](https://github.com/CoolProp/CoolProp/assets/770327/af3a50e4-65b6-45d6-a1a4-fcd4da213858)

**Comment by Rod-Persky** [comment]
@jowr : @softwareengineerprogrammer 

The DNS hosting is done with name.com if there is a need I'll look into how to delegate maintenance otherwise just let me know if/what changes are required.

Also did you want a custom email domain? There are MX records which I think are pointing to a free Google GSuite account, but in 2022 Google shut all the free GSuite accounts down (who did not transition to paying accounts).

**Comment by softwareengineerprogrammer** [comment]
@Rod-Persky I can't do anything without access to the name.com account and whatever hosting provider you have. I would start by looking for a guide on how to configure SSL for whatever server implementation coolprop.org is using and following the steps - for example, if it's Apache, this guide looks fine: https://www.digitalocean.com/community/tutorials/how-to-secure-apache-with-let-s-encrypt-on-ubuntu-20-04

**Comment by Rod-Persky** [comment]
I'm not sure where/how the website is being hosted, but hopefully @ibell or @jowr might be able to follow the guide.

Should the site be hosted on a VM the let's encrypt certbot tools will work perfectly without needing DNS access.

If we cannot get this working easily we can also use cloudflare to do the HTTPS and that will need a dns change. Cloudflare will terminate the HTTPS on their side and proxys the current server.

**Comment by ibell** [comment]
I think the only viable solution is probably to get everyone that is interested on this topic into a video chat and sit down and get it done. I think @jowr @Rod-Persky and @softwareengineerprogrammer would be a good grouping to have all the necessary info

**Comment by softwareengineerprogrammer** [comment]
> I think the only viable solution is probably to get everyone that is interested on this topic into a video chat and sit down and get it done. I think @jowr @Rod-Persky and @softwareengineerprogrammer would be a good grouping to have all the necessary info

Sounds good - feel free to get in touch with me privately via contact info in my profile for scheduling

**Comment by Rod-Persky** [comment]
To make scheduling a bit easier I've setup https://cal.com/roddersaus 😊

**Comment by Rod-Persky** [comment]
@softwareengineerprogrammer and I chatted regarding this, we're going to proxy the sourceforge via cloudflare which will enable SSL. I've also updated the existing DNS records based on the sourceforge documentation for custom domain names and confirmed that coolprop.org is online :)

**Comment by jowr** [comment]
Sorry for chiming in super late, but are we sure that this issue is fixed?

![image](https://github.com/user-attachments/assets/137e64e5-5da7-4f2f-a69e-4a2ce0edf3c0)

**Comment by fwitte** [comment]
> Sorry for chiming in super late, but are we sure that this issue is fixed?
> 
> ![image](https://private-user-images.githubusercontent.com/769593/388663737-137e64e5-5da7-4f2f-a69e-4a2ce0edf3c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNjE4MTYsIm5iZiI6MTczMjM2MTUxNiwicGF0aCI6Ii83Njk1OTMvMzg4NjYzNzM3LTEzN2U2NGU1LTVkYTctNGYyZi1hNjllLTRhMmNlMGVkZjNjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyM1QxMTMxNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hODMyZDc3NmU2OTE2ZjBlYjI1YjI5YmE3ZGNlNzFmMzc2MzIxNjBiYTBkMDhiMDUxZmYyMTg3NGQ4ZjdjZWFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0_BwuFUCKEO6TlHpfH34IbQb7RP2njjPPtOc1EoNF-Y)

I get the same warning on firefox (windows) and then get redirected to the http version when ignoring that.

**Comment by Rod-Persky** [comment]
Hey folks, we mitigated the issue where the server up address had changed in sourceforge but I'm still to setup a cloud flare account to do https termination

**Comment by softwareengineerprogrammer** [comment]
> Hey folks, we mitigated the issue where the server up address had changed in sourceforge but I'm still to setup a cloud flare account to do https termination

Thanks for clarifying @Rod-Persky - happy to jump on a call again for those steps if desired

**Comment by wta12** [comment]
I got blocked from CoolProp documentation website  with ssl secured connection (https://coolprop.org) only unsecured one work. Is there any update on this topic, thanks ? 

![Image](https://github.com/user-attachments/assets/ddbe64d6-744a-44d3-8ce5-3de50a7b9f9b)

**Comment by ibell** [comment]
In the interim while we work to get this to work perfectly, you can check
out the development docs at https://coolprop.github.io/CoolProp/

**Comment by Rod-Persky** [comment]
Just confirming progress on this, @ibell can I confirm you've setup a name.com account ending in `d5f`.

**Comment by Rod-Persky** [comment]
@ibell you will need to add A records for IPv4 users, AAAA is for IPv6. This will resolve the access issue @fwitte mentioned.

![image](https://github.com/user-attachments/assets/8b8fd149-91c4-4076-881c-dfb94b5f9b7d)

**Comment by fwitte** [comment]
> [@fwitte](https://github.com/fwitte) you're having issues as your computer is on an IPv4 only network, it may be that you are on a home or enterprise network that doesn't use IPv6.

Oh, thank you! It was the work VPN, it works after logging out! Thank you!

**Comment by ibell** [comment]
Gotcha, I'll add those records this evening

**Comment by ibell** [comment]
I think I fixed it by adding the A records indicated here: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain . Can you please try again when you have a moment?

**Comment by fwitte** [comment]
It works again for me, thank you for the quick fix!

Have a sunny day!
